### ISSUE TITLE
[Feature]: Add ability to `Get-AzVmss` on `-ResourceId`

### ISSUE THREAD

**Issue by o-l-a-v** [Azure/azure-powershell#21511]
### Description of the new feature

Please make `Get-AzVmss` (and probably other `<verb>-AzVmss` cmdlets) more flexible by adding optional `-ResourceId` parameter, like `Get-AzVm` and many other Az cmdlets already have.

Would also make it possible to `Get-AzVmss` without first having to `Set-AzContext`. Which could help make scripts faster to run, with less logic to create and maintain.

If I have resourceid as input in scripts now, I currently solve it like so:

```powershell
Get-AzVmss -ResourceGroupName $VmssResourceId.Split('/')[4] -VMScaleSetName $VmssResourceId.Split('/')[-1]
```

With this feature implemented, it can be done much cleaner like so:

```powershell
Get-AzVmss -ResourceId $VmssResourceId
```

### Proposed implementation details (optional)

Just like `Get-AzVm`.

**Comment by Drewm3** [comment]
@hilaryw29, please work with the team to see if this would be reasonable to bring in to Get-AzVMSS cmdlet. I know that it was reasonable with Get-AzVM because that cmdlet isn't auto-generated, but I am not sure if it will be as easy for Get-AzVMSS. If this cmdlet is auto-generated, then we should talk to the team that manages the auto-generation support to see about getting support for ResourceId as a general addition to the tooling.

**Comment by Sandido** [comment]
I've made a work item in our backlog for this feature. ETA is TBD.

**Comment by hilaryw29** [comment]
Hi @o-l-a-v, this feature is now available in PowerShell version 10.1.0 https://github.com/Azure/azure-powershell/releases/tag/v10.1.0-July2023. Thanks for your suggestion!
